blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/junit/BlueJUnitTrend.java

```diff
@@ -0,0 +1,150 @@
+package io.jenkins.blueocean.service.embedded.rest.junit;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Lists;
+import hudson.Extension;
+import io.jenkins.blueocean.rest.factory.BlueTrendFactory;
+import io.jenkins.blueocean.rest.hal.Link;
+import io.jenkins.blueocean.rest.model.BluePipeline;
+import io.jenkins.blueocean.rest.model.BlueRun;
+import io.jenkins.blueocean.rest.model.BlueTable;
+import io.jenkins.blueocean.rest.model.BlueTestSummary;
+import io.jenkins.blueocean.rest.model.BlueTrend;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.export.Exported;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+@Restricted(NoExternalUse.class)
+public class BlueJUnitTrend extends BlueTrend {
+
+    private static final String TOTAL = "total";
+    private static final String PASSED = "passed";
+    private static final String FIXED = "fixed";
+    private static final String FAILED = "failed";
+    private static final String EXISTING_FAILED = "existingFailed";
+    private static final String REGRESSIONS = "regressions";
+    private static final String SKIPPED = "skipped";
+
+    private final BluePipeline pipeline;
+    private final Link parent;
+
+    public BlueJUnitTrend(BluePipeline pipeline, Link parent) {
+        this.pipeline = pipeline;
+        this.parent = parent;
+    }
+
+    @Override
+    public String getId() {
+        return "junit";
+    }
+
+    @Override
+    public BlueTable getTable() {
```

Comment:
I left my comment on getRows() and getColumns() to be pageable. specially, `BlueTable.getRows()` should be pageable.

Reply:
I agree that `getRows` should be pageable, since a row tends to correspond to a run and there can easily be 100's or 1000's of them. Columns should generally be under a dozen for most use cases, so I don't think pagination is necessary there.